app/main/views.py

```diff
@@ -120,10 +131,10 @@ def update(service_id, section):
         )
 
     if form.errors:
+        service.update(form.dirty_data)
```

Comment:
@allait I've added a table to the pull request description which (hopefully) explains it.